independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -626,6 +629,10 @@ private ResourceMethod createResourceMethod(ClassInfo currentClassInfo, ClassInf
                                 "Resource method " + currentMethodInfo + " can only have a single body parameter: "
                                         + currentMethodInfo.parameterName(i));
                     bodyParamType = paramType;
+                    if (GET.equals(httpMethod) || HEAD.equals(httpMethod) || OPTIONS.equals(httpMethod)) {
+                        log.warn("Using a body parameter with " + httpMethod + " is strongly discouraged. Offending method is '"
```

Comment:
"Strongly" might be too much: I agree that Head and Options should really not have a request body, while for Get, even if unusual, some services use it to pass arguments to filter response that would not fit as headers or path parameters.
Just my 2 cents